diff --git a/chosen.js b/chosen.js
index 0450354..79f789f 100644
--- a/chosen.js
+++ b/chosen.js
@@ -30,8 +30,7 @@
             minOptions = minOptionsMultiple;
           }
 
-          options.max_selected_options = 1;
-          if (maxSelectedOptions[name] != undefined) {
+          if (maxSelectedOptions[name] != false) {
             options.max_selected_options = maxSelectedOptions[name];
           }
 
diff --git a/chosen.module b/chosen.module
index 96308bf..b213173 100644
--- a/chosen.module
+++ b/chosen.module
@@ -174,15 +174,22 @@ function chosen_pre_render_select($element) {
     $element_name = $element['#name'] . '[]';
   }
 
+  $max_selected_options = FALSE;
+  if (isset($field_info['cardinality'])) {
+    if ($field_info['cardinality'] != FIELD_CARDINALITY_UNLIMITED) {
+      $max_selected_options = $field_info['cardinality'];
+    }
+  }
+
   $element['#attached']['library'][] = array('chosen', 'drupal.chosen');
   $element['#attached']['js'][] = array(
     'data' => array(
       'chosen' => array(
         'multiple' => array(
-          $element_name => $element['#multiple'],
+          $element_name => isset($element['#multiple']) ? $element['#multiple'] : FALSE,
         ),
         'max_selected_options' => array(
-          $element_name => isset($field_info['cardinality']) ? $field_info['cardinality'] : "1",
+          $element_name => $max_selected_options,
         ),
       ),
     ),
